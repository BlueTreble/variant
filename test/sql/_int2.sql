\set ECHO 0
\set QUIET 1

/*
 * SEE NOTES IN _text.sql!
 */

\set baseline_type numeric[]
-- Space separated string
\set test_types 'int2[] int4[] int8[] float[] real[] numeric[]'
\set test_types 'int2[]'
-- Used in array[ :base_values ]
\set base_values '$${-1,-1,1}$$, $${-1,0,1}$$, $${-1,1,1}$$, $${NULL,-1,1}$$, $${NULL,0,1}$$, $${NULL,1,1}$$, $${-1,NULL,1}$$, NULL'
\set compare_value '{-1,0,1}'


\i test/helpers/type_setup.sql

SELECT plan(sum(test_count)::int) FROM plan;

\i test/helpers/type.sql


-- vi: expandtab sw=2 ts=2
