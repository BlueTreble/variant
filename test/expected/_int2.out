\set ECHO none
ok 1..0


1..174
ok 1 - No NULL variants
ok 2 - Verify result of cast '{-1,-1,1}'::smallint[]
ok 3 - Verify result of cast '{-1,0,1}'::smallint[]
ok 4 - Verify result of cast '{-1,1,1}'::smallint[]
ok 5 - Verify result of cast '{NULL,-1,1}'::smallint[]
ok 6 - Verify result of cast '{NULL,0,1}'::smallint[]
ok 7 - Verify result of cast '{NULL,1,1}'::smallint[]
ok 8 - Verify result of cast '{-1,NULL,1}'::smallint[]
ok 9 - Verify result of cast NULL::smallint[]
ok 10 - SET base_variant_b
ok 11 - Verify text_in(text_out()) works for (smallint[],"{-1,-1,1}")
ok 12 - Verify text_in(text_out()) works for (smallint[],"{-1,0,1}")
ok 13 - Verify text_in(text_out()) works for (smallint[],"{-1,1,1}")
ok 14 - Verify text_in(text_out()) works for (smallint[],"{NULL,-1,1}")
ok 15 - Verify text_in(text_out()) works for (smallint[],"{NULL,0,1}")
ok 16 - Verify text_in(text_out()) works for (smallint[],"{NULL,1,1}")
ok 17 - Verify text_in(text_out()) works for (smallint[],"{-1,NULL,1}")
ok 18 - Verify text_in(text_out()) works for (smallint[],)
ok 19
ok 20
ok 21
ok 22
ok 23
ok 24
ok 25
ok 26
ok 27 - No NULL variants
ok 28 - Verify result of cast '{-1,0,1}'::smallint[]
ok 29 - SET compare_variant_b
ok 30 - Verify text_in(text_out()) works for (smallint[],"{-1,0,1}")
ok 31 - check casted vs uncasted for '{-1,0,1}'::smallint[] <   '{-1,-1,1}'::smallint[]
ok 32 - check casted vs uncasted for '{-1,0,1}'::smallint[] <   '{-1,0,1}'::smallint[]
ok 33 - check casted vs uncasted for '{-1,0,1}'::smallint[] <   '{-1,1,1}'::smallint[]
ok 34 - check casted vs uncasted for '{-1,0,1}'::smallint[] <   '{-1,NULL,1}'::smallint[]
ok 35 - check casted vs uncasted for '{-1,0,1}'::smallint[] <   '{NULL,-1,1}'::smallint[]
ok 36 - check casted vs uncasted for '{-1,0,1}'::smallint[] <   '{NULL,0,1}'::smallint[]
ok 37 - check casted vs uncasted for '{-1,0,1}'::smallint[] <   '{NULL,1,1}'::smallint[]
ok 38 - check casted vs uncasted for '{-1,0,1}'::smallint[] <   NULL::smallint[]
ok 39 - check casted vs uncasted for '{-1,0,1}'::smallint[] <=  '{-1,-1,1}'::smallint[]
ok 40 - check casted vs uncasted for '{-1,0,1}'::smallint[] <=  '{-1,0,1}'::smallint[]
ok 41 - check casted vs uncasted for '{-1,0,1}'::smallint[] <=  '{-1,1,1}'::smallint[]
ok 42 - check casted vs uncasted for '{-1,0,1}'::smallint[] <=  '{-1,NULL,1}'::smallint[]
ok 43 - check casted vs uncasted for '{-1,0,1}'::smallint[] <=  '{NULL,-1,1}'::smallint[]
ok 44 - check casted vs uncasted for '{-1,0,1}'::smallint[] <=  '{NULL,0,1}'::smallint[]
ok 45 - check casted vs uncasted for '{-1,0,1}'::smallint[] <=  '{NULL,1,1}'::smallint[]
ok 46 - check casted vs uncasted for '{-1,0,1}'::smallint[] <=  NULL::smallint[]
ok 47 - check casted vs uncasted for '{-1,0,1}'::smallint[]  =  '{-1,-1,1}'::smallint[]
ok 48 - check casted vs uncasted for '{-1,0,1}'::smallint[]  =  '{-1,0,1}'::smallint[]
ok 49 - check casted vs uncasted for '{-1,0,1}'::smallint[]  =  '{-1,1,1}'::smallint[]
ok 50 - check casted vs uncasted for '{-1,0,1}'::smallint[]  =  '{-1,NULL,1}'::smallint[]
ok 51 - check casted vs uncasted for '{-1,0,1}'::smallint[]  =  '{NULL,-1,1}'::smallint[]
ok 52 - check casted vs uncasted for '{-1,0,1}'::smallint[]  =  '{NULL,0,1}'::smallint[]
ok 53 - check casted vs uncasted for '{-1,0,1}'::smallint[]  =  '{NULL,1,1}'::smallint[]
ok 54 - check casted vs uncasted for '{-1,0,1}'::smallint[]  =  NULL::smallint[]
ok 55 - check casted vs uncasted for '{-1,0,1}'::smallint[]  >= '{-1,-1,1}'::smallint[]
ok 56 - check casted vs uncasted for '{-1,0,1}'::smallint[]  >= '{-1,0,1}'::smallint[]
ok 57 - check casted vs uncasted for '{-1,0,1}'::smallint[]  >= '{-1,1,1}'::smallint[]
ok 58 - check casted vs uncasted for '{-1,0,1}'::smallint[]  >= '{-1,NULL,1}'::smallint[]
ok 59 - check casted vs uncasted for '{-1,0,1}'::smallint[]  >= '{NULL,-1,1}'::smallint[]
ok 60 - check casted vs uncasted for '{-1,0,1}'::smallint[]  >= '{NULL,0,1}'::smallint[]
ok 61 - check casted vs uncasted for '{-1,0,1}'::smallint[]  >= '{NULL,1,1}'::smallint[]
ok 62 - check casted vs uncasted for '{-1,0,1}'::smallint[]  >= NULL::smallint[]
ok 63 - check casted vs uncasted for '{-1,0,1}'::smallint[]   > '{-1,-1,1}'::smallint[]
ok 64 - check casted vs uncasted for '{-1,0,1}'::smallint[]   > '{-1,0,1}'::smallint[]
ok 65 - check casted vs uncasted for '{-1,0,1}'::smallint[]   > '{-1,1,1}'::smallint[]
ok 66 - check casted vs uncasted for '{-1,0,1}'::smallint[]   > '{-1,NULL,1}'::smallint[]
ok 67 - check casted vs uncasted for '{-1,0,1}'::smallint[]   > '{NULL,-1,1}'::smallint[]
ok 68 - check casted vs uncasted for '{-1,0,1}'::smallint[]   > '{NULL,0,1}'::smallint[]
ok 69 - check casted vs uncasted for '{-1,0,1}'::smallint[]   > '{NULL,1,1}'::smallint[]
ok 70 - check casted vs uncasted for '{-1,0,1}'::smallint[]   > NULL::smallint[]
ok 71 - check casted vs uncasted for '{-1,0,1}'::smallint[] !=  '{-1,-1,1}'::smallint[]
ok 72 - check casted vs uncasted for '{-1,0,1}'::smallint[] !=  '{-1,0,1}'::smallint[]
ok 73 - check casted vs uncasted for '{-1,0,1}'::smallint[] !=  '{-1,1,1}'::smallint[]
ok 74 - check casted vs uncasted for '{-1,0,1}'::smallint[] !=  '{-1,NULL,1}'::smallint[]
ok 75 - check casted vs uncasted for '{-1,0,1}'::smallint[] !=  '{NULL,-1,1}'::smallint[]
ok 76 - check casted vs uncasted for '{-1,0,1}'::smallint[] !=  '{NULL,0,1}'::smallint[]
ok 77 - check casted vs uncasted for '{-1,0,1}'::smallint[] !=  '{NULL,1,1}'::smallint[]
ok 78 - check casted vs uncasted for '{-1,0,1}'::smallint[] !=  NULL::smallint[]
ok 79 - check variant vs uncasted for '{-1,0,1}'::smallint[] <   '{-1,-1,1}'::smallint[]
ok 80 - check variant vs uncasted for '{-1,0,1}'::smallint[] <   '{-1,0,1}'::smallint[]
ok 81 - check variant vs uncasted for '{-1,0,1}'::smallint[] <   '{-1,1,1}'::smallint[]
ok 82 - check variant vs uncasted for '{-1,0,1}'::smallint[] <   '{-1,NULL,1}'::smallint[]
ok 83 - check variant vs uncasted for '{-1,0,1}'::smallint[] <   '{NULL,-1,1}'::smallint[]
ok 84 - check variant vs uncasted for '{-1,0,1}'::smallint[] <   '{NULL,0,1}'::smallint[]
ok 85 - check variant vs uncasted for '{-1,0,1}'::smallint[] <   '{NULL,1,1}'::smallint[]
ok 86 - check variant vs uncasted for '{-1,0,1}'::smallint[] <   NULL::smallint[]
ok 87 - check variant vs uncasted for '{-1,0,1}'::smallint[] <=  '{-1,-1,1}'::smallint[]
ok 88 - check variant vs uncasted for '{-1,0,1}'::smallint[] <=  '{-1,0,1}'::smallint[]
ok 89 - check variant vs uncasted for '{-1,0,1}'::smallint[] <=  '{-1,1,1}'::smallint[]
ok 90 - check variant vs uncasted for '{-1,0,1}'::smallint[] <=  '{-1,NULL,1}'::smallint[]
ok 91 - check variant vs uncasted for '{-1,0,1}'::smallint[] <=  '{NULL,-1,1}'::smallint[]
ok 92 - check variant vs uncasted for '{-1,0,1}'::smallint[] <=  '{NULL,0,1}'::smallint[]
ok 93 - check variant vs uncasted for '{-1,0,1}'::smallint[] <=  '{NULL,1,1}'::smallint[]
ok 94 - check variant vs uncasted for '{-1,0,1}'::smallint[] <=  NULL::smallint[]
ok 95 - check variant vs uncasted for '{-1,0,1}'::smallint[]  =  '{-1,-1,1}'::smallint[]
ok 96 - check variant vs uncasted for '{-1,0,1}'::smallint[]  =  '{-1,0,1}'::smallint[]
ok 97 - check variant vs uncasted for '{-1,0,1}'::smallint[]  =  '{-1,1,1}'::smallint[]
ok 98 - check variant vs uncasted for '{-1,0,1}'::smallint[]  =  '{-1,NULL,1}'::smallint[]
ok 99 - check variant vs uncasted for '{-1,0,1}'::smallint[]  =  '{NULL,-1,1}'::smallint[]
ok 100 - check variant vs uncasted for '{-1,0,1}'::smallint[]  =  '{NULL,0,1}'::smallint[]
ok 101 - check variant vs uncasted for '{-1,0,1}'::smallint[]  =  '{NULL,1,1}'::smallint[]
ok 102 - check variant vs uncasted for '{-1,0,1}'::smallint[]  =  NULL::smallint[]
ok 103 - check variant vs uncasted for '{-1,0,1}'::smallint[]  >= '{-1,-1,1}'::smallint[]
ok 104 - check variant vs uncasted for '{-1,0,1}'::smallint[]  >= '{-1,0,1}'::smallint[]
ok 105 - check variant vs uncasted for '{-1,0,1}'::smallint[]  >= '{-1,1,1}'::smallint[]
ok 106 - check variant vs uncasted for '{-1,0,1}'::smallint[]  >= '{-1,NULL,1}'::smallint[]
ok 107 - check variant vs uncasted for '{-1,0,1}'::smallint[]  >= '{NULL,-1,1}'::smallint[]
ok 108 - check variant vs uncasted for '{-1,0,1}'::smallint[]  >= '{NULL,0,1}'::smallint[]
ok 109 - check variant vs uncasted for '{-1,0,1}'::smallint[]  >= '{NULL,1,1}'::smallint[]
ok 110 - check variant vs uncasted for '{-1,0,1}'::smallint[]  >= NULL::smallint[]
ok 111 - check variant vs uncasted for '{-1,0,1}'::smallint[]   > '{-1,-1,1}'::smallint[]
ok 112 - check variant vs uncasted for '{-1,0,1}'::smallint[]   > '{-1,0,1}'::smallint[]
ok 113 - check variant vs uncasted for '{-1,0,1}'::smallint[]   > '{-1,1,1}'::smallint[]
ok 114 - check variant vs uncasted for '{-1,0,1}'::smallint[]   > '{-1,NULL,1}'::smallint[]
ok 115 - check variant vs uncasted for '{-1,0,1}'::smallint[]   > '{NULL,-1,1}'::smallint[]
ok 116 - check variant vs uncasted for '{-1,0,1}'::smallint[]   > '{NULL,0,1}'::smallint[]
ok 117 - check variant vs uncasted for '{-1,0,1}'::smallint[]   > '{NULL,1,1}'::smallint[]
ok 118 - check variant vs uncasted for '{-1,0,1}'::smallint[]   > NULL::smallint[]
ok 119 - check variant vs uncasted for '{-1,0,1}'::smallint[] !=  '{-1,-1,1}'::smallint[]
ok 120 - check variant vs uncasted for '{-1,0,1}'::smallint[] !=  '{-1,0,1}'::smallint[]
ok 121 - check variant vs uncasted for '{-1,0,1}'::smallint[] !=  '{-1,1,1}'::smallint[]
ok 122 - check variant vs uncasted for '{-1,0,1}'::smallint[] !=  '{-1,NULL,1}'::smallint[]
ok 123 - check variant vs uncasted for '{-1,0,1}'::smallint[] !=  '{NULL,-1,1}'::smallint[]
ok 124 - check variant vs uncasted for '{-1,0,1}'::smallint[] !=  '{NULL,0,1}'::smallint[]
ok 125 - check variant vs uncasted for '{-1,0,1}'::smallint[] !=  '{NULL,1,1}'::smallint[]
ok 126 - check variant vs uncasted for '{-1,0,1}'::smallint[] !=  NULL::smallint[]
ok 127 - check variant vs casted for '''{-1,0,1}''::smallint[]' <   '''{-1,-1,1}''::smallint[]'
ok 128 - check variant vs casted for '''{-1,0,1}''::smallint[]' <   '''{-1,0,1}''::smallint[]'
ok 129 - check variant vs casted for '''{-1,0,1}''::smallint[]' <   '''{-1,1,1}''::smallint[]'
ok 130 - check variant vs casted for '''{-1,0,1}''::smallint[]' <   '''{-1,NULL,1}''::smallint[]'
ok 131 - check variant vs casted for '''{-1,0,1}''::smallint[]' <   '''{NULL,-1,1}''::smallint[]'
ok 132 - check variant vs casted for '''{-1,0,1}''::smallint[]' <   '''{NULL,0,1}''::smallint[]'
ok 133 - check variant vs casted for '''{-1,0,1}''::smallint[]' <   '''{NULL,1,1}''::smallint[]'
ok 134 - check variant vs casted for '''{-1,0,1}''::smallint[]' <   'NULL::smallint[]'
ok 135 - check variant vs casted for '''{-1,0,1}''::smallint[]' <=  '''{-1,-1,1}''::smallint[]'
ok 136 - check variant vs casted for '''{-1,0,1}''::smallint[]' <=  '''{-1,0,1}''::smallint[]'
ok 137 - check variant vs casted for '''{-1,0,1}''::smallint[]' <=  '''{-1,1,1}''::smallint[]'
ok 138 - check variant vs casted for '''{-1,0,1}''::smallint[]' <=  '''{-1,NULL,1}''::smallint[]'
ok 139 - check variant vs casted for '''{-1,0,1}''::smallint[]' <=  '''{NULL,-1,1}''::smallint[]'
ok 140 - check variant vs casted for '''{-1,0,1}''::smallint[]' <=  '''{NULL,0,1}''::smallint[]'
ok 141 - check variant vs casted for '''{-1,0,1}''::smallint[]' <=  '''{NULL,1,1}''::smallint[]'
ok 142 - check variant vs casted for '''{-1,0,1}''::smallint[]' <=  'NULL::smallint[]'
ok 143 - check variant vs casted for '''{-1,0,1}''::smallint[]'  =  '''{-1,-1,1}''::smallint[]'
ok 144 - check variant vs casted for '''{-1,0,1}''::smallint[]'  =  '''{-1,0,1}''::smallint[]'
ok 145 - check variant vs casted for '''{-1,0,1}''::smallint[]'  =  '''{-1,1,1}''::smallint[]'
ok 146 - check variant vs casted for '''{-1,0,1}''::smallint[]'  =  '''{-1,NULL,1}''::smallint[]'
ok 147 - check variant vs casted for '''{-1,0,1}''::smallint[]'  =  '''{NULL,-1,1}''::smallint[]'
ok 148 - check variant vs casted for '''{-1,0,1}''::smallint[]'  =  '''{NULL,0,1}''::smallint[]'
ok 149 - check variant vs casted for '''{-1,0,1}''::smallint[]'  =  '''{NULL,1,1}''::smallint[]'
ok 150 - check variant vs casted for '''{-1,0,1}''::smallint[]'  =  'NULL::smallint[]'
ok 151 - check variant vs casted for '''{-1,0,1}''::smallint[]'  >= '''{-1,-1,1}''::smallint[]'
ok 152 - check variant vs casted for '''{-1,0,1}''::smallint[]'  >= '''{-1,0,1}''::smallint[]'
ok 153 - check variant vs casted for '''{-1,0,1}''::smallint[]'  >= '''{-1,1,1}''::smallint[]'
ok 154 - check variant vs casted for '''{-1,0,1}''::smallint[]'  >= '''{-1,NULL,1}''::smallint[]'
ok 155 - check variant vs casted for '''{-1,0,1}''::smallint[]'  >= '''{NULL,-1,1}''::smallint[]'
ok 156 - check variant vs casted for '''{-1,0,1}''::smallint[]'  >= '''{NULL,0,1}''::smallint[]'
ok 157 - check variant vs casted for '''{-1,0,1}''::smallint[]'  >= '''{NULL,1,1}''::smallint[]'
ok 158 - check variant vs casted for '''{-1,0,1}''::smallint[]'  >= 'NULL::smallint[]'
ok 159 - check variant vs casted for '''{-1,0,1}''::smallint[]'   > '''{-1,-1,1}''::smallint[]'
ok 160 - check variant vs casted for '''{-1,0,1}''::smallint[]'   > '''{-1,0,1}''::smallint[]'
ok 161 - check variant vs casted for '''{-1,0,1}''::smallint[]'   > '''{-1,1,1}''::smallint[]'
ok 162 - check variant vs casted for '''{-1,0,1}''::smallint[]'   > '''{-1,NULL,1}''::smallint[]'
ok 163 - check variant vs casted for '''{-1,0,1}''::smallint[]'   > '''{NULL,-1,1}''::smallint[]'
ok 164 - check variant vs casted for '''{-1,0,1}''::smallint[]'   > '''{NULL,0,1}''::smallint[]'
ok 165 - check variant vs casted for '''{-1,0,1}''::smallint[]'   > '''{NULL,1,1}''::smallint[]'
ok 166 - check variant vs casted for '''{-1,0,1}''::smallint[]'   > 'NULL::smallint[]'
ok 167 - check variant vs casted for '''{-1,0,1}''::smallint[]' !=  '''{-1,-1,1}''::smallint[]'
ok 168 - check variant vs casted for '''{-1,0,1}''::smallint[]' !=  '''{-1,0,1}''::smallint[]'
ok 169 - check variant vs casted for '''{-1,0,1}''::smallint[]' !=  '''{-1,1,1}''::smallint[]'
ok 170 - check variant vs casted for '''{-1,0,1}''::smallint[]' !=  '''{-1,NULL,1}''::smallint[]'
ok 171 - check variant vs casted for '''{-1,0,1}''::smallint[]' !=  '''{NULL,-1,1}''::smallint[]'
ok 172 - check variant vs casted for '''{-1,0,1}''::smallint[]' !=  '''{NULL,0,1}''::smallint[]'
ok 173 - check variant vs casted for '''{-1,0,1}''::smallint[]' !=  '''{NULL,1,1}''::smallint[]'
ok 174 - check variant vs casted for '''{-1,0,1}''::smallint[]' !=  'NULL::smallint[]'
